akka-projection-poc2/src/main/scala/akka/projection/eventsourced/jdbc/Demo.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.projection.eventsourced.jdbc
+
+import java.sql.Connection
+
+import scala.concurrent.Future
+
+import akka.Done
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+
+object Demo {
+
+  object ShoppingCart {
+    trait Event
+  }
+
+  object ShoppingCartProjection {
+    def start(system: ActorSystem[_]): Unit = {
+
+      val eventProcessorId = "ShoppingCartProcessor"
+      val tag = "CartSlice-1"
+
+      implicit val ec = system.executionContext
+
+      val projectionHandler: JdbcSingleEventHandlerWithTxOffset[ShoppingCart.Event] =
+        new JdbcSingleEventHandlerWithTxOffset[ShoppingCart.Event](eventProcessorId, tag) {
+          override def getConnection(): Future[Connection] = ???
+
+          override def onEvent(event: ShoppingCart.Event): Future[Done] = {
+            for {
+              c <- getConnection()
+              _ <- saveEventProjection(c, event)
+              _ <- saveOffset(c) // in same tx
```

Comment:
In this example the saveOffset is implemented by Akka, but I can imagine completely different ways of storing the offset together with the user data. That would also be supported by the `OffsetManagedByProjectionHandler`.

Reply:
Maybe what we should do is remove the runner and only have an optional OffsetManager/Strategy. 

If a user prefers to manage the offset, they can build their own and pass it to the handler. 

The handler can then receive the Envelope (a matter of tweaking the extractor).

Reply:
On another note, I'm worried about leaving too much to be managed on user level. We have three Futures here. So we will have a connection (and transaction) being shared by potentially three threads. 

Also, when will the connection be returned to the pool? By whom? 

My initial idea of the IO type (that I later renamed to Result to avoid any association with IO monad) was to not necessarily make it a monad. 

For Slick, it would be DBIO and we could simply flatMap and execute transactionally. 
For pure JDBC or JPA, we could have it typed to `Done`. Then, the runner or offset strategy would take care of it, something like: 

```scala
withTransaction { connec => 
	handler.onEvent(connec, evt) // returns Done
	saveOffset(connec, offset) // returns Done
}  // returns Future[Done]
```

In this example, we manage the connection and the thread pool in which the blocking JDBC will be executed. 

The event handle needs to be specific to JDBC or JPA, because it needs the connection to be passed. This is not much different then what Lagom does for instance.